components/raftstore/src/store/peer_storage.rs

```diff
@@ -422,28 +406,28 @@ impl From<Error> for RaftError {
     }
 }
 
-pub struct ApplySnapResult {
-    // prev_region is the region before snapshot applied.
-    pub prev_region: metapb::Region,
-    pub region: metapb::Region,
-    pub destroyed_regions: Vec<metapb::Region>,
+#[derive(PartialEq, Debug)]
+pub enum HandleReadyResult {
+    SendIOTask,
+    Snapshot {
+        msgs: Vec<RaftMessage>,
+        snap_region: metapb::Region,
+        /// The regions whose range are overlapped with this region
+        destroy_regions: Vec<Region>,
+    },
+    NoIOTask {
+        msgs: Vec<RaftMessage>,
+    },
 }
 
-/// Returned by `PeerStorage::handle_raft_ready`, used for recording changed status of
-/// `RaftLocalState` and `RaftApplyState`.
+/// Used for recording changed status of `RaftLocalState` and `RaftApplyState`.
 pub struct InvokeContext {
```

Comment:
Now all states are updated in place without writing to disk, the struct seems can be deleted.